core/src/main/java/com/novoda/noplayer/drm/DrmSessionCreatorFactory.java

```diff
@@ -1,27 +1,66 @@
 package com.novoda.noplayer.drm;
 
+import android.os.Build;
+import android.os.Handler;
+
+import com.novoda.noplayer.drm.provision.HttpPoster;
 import com.novoda.noplayer.drm.provision.ProvisionExecutor;
+import com.novoda.noplayer.drm.provision.ProvisioningCapabilities;
 import com.novoda.noplayer.player.PlayerFactory;
+import com.novoda.utils.AndroidDeviceVersion;
 
 public class DrmSessionCreatorFactory {
 
+    private final AndroidDeviceVersion androidDeviceVersion;
+    private final Handler handler;
+
+    public DrmSessionCreatorFactory(AndroidDeviceVersion androidDeviceVersion, Handler handler) {
+        this.androidDeviceVersion = androidDeviceVersion;
+        this.handler = handler;
+    }
+
     public DrmSessionCreator createFor(DrmType drmType, DrmHandler drmHandler) {
```

Comment:
nope, they'll be thrown if the device or available players can't handle the drm types (content types should eventually be taken into account here as well)